src/main/java/com/robomwm/prettysimpleshop/feature/ShowoffItem.java

```diff
@@ -184,19 +184,23 @@ private void onDoubleChest(BlockPlaceEvent event)
             return;
         if (!config.isShopBlock(event.getBlock().getType()))
             return;
-        new BukkitRunnable()
-        {
+
+        if(event.getBlockPlaced().getType() != Material.CHEST && event.getBlockPlaced().getType() != Material.TRAPPED_CHEST)
+            return;
+
+        new BukkitRunnable() {
             @Override
-            public void run()
-            {
-                InventoryHolder holder = ((Container)event.getBlock().getState()).getInventory().getHolder();
+            public void run() {
+                Container container = shopAPI.getContainer(event.getBlock().getLocation());
+                InventoryHolder holder = container.getInventory().getHolder();
                 if (!(holder instanceof DoubleChest))
                     return;
                 DoubleChest doubleChest = (DoubleChest)holder;
                 despawnItem(((Chest)(doubleChest.getLeftSide())).getLocation().add(0.5, 1.2, 0.5));
-                despawnItem(((Chest)(doubleChest.getLeftSide())).getLocation().add(0.5, 1.2, 0.5));
+                despawnItem(((Chest)(doubleChest.getRightSide())).getLocation().add(0.5, 1.2, 0.5));
+                spawnItem(new ShopInfo(shopAPI.getLocation(container), shopAPI.getItemStack(container), shopAPI.getPrice(container)));
             }
-        }.runTask(plugin);
+        }.runTaskLater(plugin, 1L);
```

Comment:
What's up with the runnable here, are doublechests not being properly handled anymore?

Also, `runTask(plugin)` is equivalent to `runTaskLater(plugin, 1)`

Reply:
didnt seem to be ever handled for when they were made or broke

Reply:
Ok, so what you actually mean is you respawn the shop item when the doublechest is created instead of just having it despawn (I guess I was under the impression that I had another method cover this - that, or it's that I wasn't able to determine if the new doublechest is supposed to be a shop or not. The left side overrides, so the resulting doublechest may not be a shop, and thus the newly-spawned item won't be able to be despawned until the chunk is unloaded.)